src/app/unlocked-stage/unlocked-stage.component.ts

```diff
@@ -1,156 +1,331 @@
+              public snackBar: MatSnackBar) {
+  }
 
-    attack() {
-        this._spray.src = '../assets/img/unlocked-stage/spray.gif';
-        this._mousedown = true;
+  // If we need keyboard movement for the net
+  @HostListener('window:keyup', ['$event'])
+  keyEvent(event: KeyboardEvent) {
+
+    if (event.keyCode === 38) {
+      this.keydown = true;
+      const that = this;
+      setInterval(function () {
+        if (that.keydown) {
+          that.$net.css({top: '-=2px'});
+        }
+      }, 10);
     }
+    if (event.keyCode === 40) {
+      this.keydown = false;
+      this.$net.css({top: '+=20px'});
+    }
+  }
+
+
+  ngOnInit() {
+    this._langService.loadLanguage().subscribe(response => {
+      this.language = response.pcprepkit.stages.lockedStage.game;
+    });
+    this._canvas = document.getElementById('scene');
+
+    // Initialising single mosquito
+    this.$mosquito = $('<div><img src="../assets/img/unlocked-stage/mosquito.gif" class="moz"></div>');
+    const progress = $('<div class="mosquitoHealth">' + 100 + '</div>');
+    this.$mosquito.attr({'class': 'moz'});
+    this.$mosquito.append(progress);
 
-    stopAttack() {
-        this._spray.src = '../assets/img/unlocked-stage/spray-cursor.gif';
-        this._mousedown = false;
+    // Adding n = mosCountInitial initially in the game
+    for (let i = 1; i <= this.mosCountInitial; i++) {
+      const mosquitoClone = this.$mosquito.clone();
+      $('#scene').append(mosquitoClone);
+      this.mosquitoes.push(mosquitoClone);
     }
 
-    moveOverCanvas(e) {
-        this._canvas.style.cursor = 'none';
-        this._spray.style.display = 'block';
-        this._spray.style.left = e.pageX - 425 + 'px';
-        this._spray.style.top = e.pageY - 150 + 'px';
+    // Spray element
+    this.$spray = $('<img src="../assets/img/unlocked-stage/spray-cursor.gif">');
+    this.$spray.attr({'id': 'spray'});
+    $('#scene').append(this.$spray);
+
+    // Net element
+    this.$net = $('<img src="../assets/img/unlocked-stage/net.png">');
+    this.$net.attr({'id': 'net'});
+    $('#scene').append(this.$net);
+    this.$net.css({display: 'block'});
+    this.$net.css({left: '840px'});
+
+    // Progress and HealthBars
+    this._humanProgBar = document.getElementById('human-health-bar');
+
+    // Human
+    this.$human = $('<img src="../assets/img/unlocked-stage/human.png">').attr({'id': 'human'});
+    $('#scene').append(this.$human);
+
+    // Time increment begins as the game starts
+    this.incrementTime();
+    this._spray = this.$spray;
+    this._mousemoveListener = this._renderer.listen(this._canvas, 'mousemove', (event) => this.moveOverCanvasSpray(event));
+    this._mouseleaveListener = this._renderer.listen(this._canvas, 'mouseleave', (event) => this.leaveCanvas());
+    this._mousedownListener = this._renderer.listen(this._canvas, 'mousedown', (event) => this.attack());
```

Comment:
I think this mousedown event listener should have a check to see if the spray is exhausted before attacking because right now, the spray can be used even when it is used up.

Reply:
Done.